ibis/backends/sql/rewrites.py

```diff
@@ -52,6 +52,9 @@ class Select(ops.Relation):
     predicates: VarTuple[ops.Value[dt.Boolean]] = ()
     sort_keys: VarTuple[ops.SortKey] = ()
 
+    def is_star_selection(self):
+        return tuple(self.values.items()) == tuple(self.parent.fields.items())
```

Comment:
Use `tuple()` to also check for projection order.

Reply:
Worth caching this?

Reply:
Nope, it is only used during compilation which happens once.